src/DETHRACE/common/spark.c

```diff
@@ -873,7 +874,13 @@ void SmokeLine(int l, int x, br_scalar zbuff, int r_squared, tU8* scr_ptr, tU16*
 
     for (i = 0; i < l; i++) {
         if (*depth_ptr > z) {
-            *scr_ptr = shade_ptr[*scr_ptr + (((shade_offset_int - r_squared * r_multiplier_int) >> 8) & 0xffffff00)];
+            offset = ((shade_offset_int - r_squared * r_multiplier_int) >> 8) &
+		     0xffffff00;
+#if defined(DETHRACE_FIX_BUGS)
+            /* Prevent buffer underflows by capping negative offsets. */
+            offset = offset < 0 ? 0 : offset;
```

Comment:
At this point, we're bike shedding :)
```suggestion
            offset = MAX(0, offset);
```

Reply:
I don't mind. A good patch is a clean patch.